pkg/controllers/sync/dispatch/unmanaged.go

```diff
@@ -190,20 +192,20 @@ func (d *unmanagedDispatcherImpl) RemoveManagedLabel(clusterName string, cluster
 		if _, err := removeRetainObjectFinalizer(updateObj); err != nil {
 			if d.recorder == nil {
 				wrappedErr := d.wrapOperationError(err, clusterName, op)
-				runtime.HandleError(wrappedErr)
+				keyedLogger.Error(wrappedErr, "Failed to remove managed label from target object in cluster")
 			} else {
-				d.recorder.recordOperationError(fedtypesv1a1.LabelRemovalFailed, clusterName, op, err)
+				d.recorder.recordOperationError(ctx, fedtypesv1a1.LabelRemovalFailed, clusterName, op, err)
 			}
 			return false
 		}
 
-		err := client.Update(context.Background(), updateObj)
+		err := client.Update(ctx, updateObj)
 		if err != nil {
 			if d.recorder == nil {
 				wrappedErr := d.wrapOperationError(err, clusterName, op)
-				runtime.HandleError(wrappedErr)
+				keyedLogger.Error(wrappedErr, "Failed to remove managed label from target object in cluster")
```

Comment:
`"cluster-name"` is missed here. Let's create a sub-logger with `"cluster-name"` at the start of this function.